inst/staticexports/package.R

```diff
@@ -7,6 +7,14 @@ is_installed <- function(pkg, version = NULL) {
   if (is.null(version)) {
     return(installed)
   }
+  if (!inherits(version, c("package_version", "numeric_version"))) {
+    if (!is.character(version)) {
+      warning("`version` must be a character string or a `package_version` or `numeric_version` object.")
+      opts <- options(OutDec = ".")
+      on.exit(options(opts))
+      version <- numeric_version(as.character(version))
+    }
+  }
```

Comment:
Some suggestions for this code:
- Use `is.character()` first, because it's faster, as @schloerke noted in #27. (Speed is a high priority for functions in staticimports.)
- Instead of nested `if` statements, use `&&`. (Putting `is.character()` first still has a speed advantage due to the short-circuit behavior of `&&`.)
- No need to check for `package_version` class, because those objects are also `numeric_version` objects.
- Use `sprintf()` to control formatting instead of setting the global `OutDec` option and calling `as.character()`.

```suggestion
  if (!is.character(version) && !inherits(version, "numeric_version")) {
    warning("`version` must be a character string or `numeric_version` object.")
    version <- numeric_version(sprintf("%0.9g", version))
  }
```

Reply:
Refactored in d42f149e94bdeb1b69abec0560a4ee358bb4a0c4.